### ISSUE TITLE
consulagent service discovery mistakenly conflates service name and service ID

### ISSUE THREAD

**Issue by gautamg795** [VictoriaMetrics/VictoriaMetrics#4390]
### Describe the bug

When using consulagent service discovery (with a base/empty configuration, i.e. find all services, no filters), the discovery attempts to "watch" using the service's ID rather than the name. These fields are often identical, but are not necessarily. For example, when services are registered by Nomad, the ID field is a long programmatic name related to the running task, whereas the Name field is the service name. 

For example, given the following response from the `v1/agent/services` API:
```
{
  "_nomad-client-yafy2vcf4ssnil25kdvl3ookr4jdllxh": {
    "ID": "_nomad-client-yafy2vcf4ssnil25kdvl3ookr4jdllxh",
    "Service": "nomad-client",
    "Tags": [
      "http"
    ],
    "Meta": {
      "external-source": "nomad"
    },
    "Port": 4646,
    "Address": "10.1.120.169",
    "TaggedAddresses": {
      "lan_ipv4": {
        "Address": "10.1.120.169",
        "Port": 4646
      },
      "wan_ipv4": {
        "Address": "10.1.120.169",
        "Port": 4646
      }
    },
    "Weights": {
      "Passing": 1,
      "Warning": 1
    },
    "EnableTagOverride": false,
    "Datacenter": "us-east-1"
  },
}
```
the consulagent discovery will fail with the following error:
```
2023-06-02T05:30:44.010Z	error	VictoriaMetrics/lib/promscrape/discovery/consulagent/watch.go:231	cannot obtain Consul Agent serviceNodes for serviceName="_nomad-client-yafy2vcf4ssnil25kdvl3ookr4jdllxh" from "http://localhost:8500": unexpected status code returned from "http://localhost:8500/v1/agent/health/service/name/_nomad-client-yafy2vcf4ssnil25kdvl3ookr4jdllxh": 404; expecting 200; response body: "[]"
```
Because it is trying to find a service by name but passing the ID, it fails and displays this error.

### To Reproduce

Add a service to Consul such that the ID and Name fields are not equal, and allow consulagent to discover the service.
Running with the following config:
```
scrape_configs:
  - job_name: consulagent
    consulagent_sd_configs:
      - {}
```

### Version

vmagent-20230601-144219-tags-v1.91.1-0-g2c876227e

**Comment by gautamg795** [comment]
Thanks for the quick fix! I'll close this as your PR has been merged.

**Comment by f41gh7** [comment]
Was fixed at [v1.91.3](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.91.3) release
